src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt

```diff
@@ -331,6 +345,20 @@ class RsErrorAnnotator : Annotator, HighlightRangeExtension {
     private fun pluralise(count: Int, singular: String, plural: String): String =
         if (count == 1) singular else plural
 
+    private fun canElideReturnLifetime(fn: PsiElement): Boolean {
```

Comment:
I also wonder what would be a reliable way to catch such false positives? I always run annotations against several relatively large projects like cargo, hyper etc. But that's obviously not enough. @mkaput suggested #896, but a) I didn't figure out how to implement it, and b) I'm not sure if it'll be enough because rust code allows very convoluted code structures that people don't use in real code.

Maybe we could check annotations against some tests that rust uses? Some (chosen set of) files from [rust/test](https://github.com/rust-lang/rust/tree/b1363a73ede57ae595f3a1be2bb75d308ba4f7f6/src/test)?